rest/oidc_api.go

```diff
@@ -82,7 +107,13 @@ func (h *handler) handleOIDCCommon() (redirectURLString string, err error) {
 	}
 
 	var redirectURL *url.URL
-	state := ""
+	var state string
+
+	// Set state parameter to prevent cross-site request forgery (CSRF) when DisableCallbackState is not enabled.
+	if provider.DisableCallbackState != nil && !*provider.DisableCallbackState {
+		state = base.GenerateRandomSecret()
+		setStateCookie(h.response, state)
```

Comment:
Not sure it's nessesary to declare the variable outside of the scope here.

```suggestion
	// Set state parameter to prevent cross-site request forgery (CSRF) when DisableCallbackState is not enabled.
	if provider.DisableCallbackState != nil && !*provider.DisableCallbackState {
		setStateCookie(h.response, base.GenerateRandomSecret())
```

Reply:
It's reasonable to keep the scope of the `state` variable outside the if block that checks whether `DisableCallbackState` is enabled/disabled because we need not only to set the state in cookie but also include in the authorization request URL whilst making auth request against OAuth server and get to it returned back in the callbackURL on an approved user consent. Also keeping the scope of `state` variable outside the if block gracefully handles the scenario where the state doesn't require to maintain between SG and auth server; the `AuthCodeURL` method always accept state parameter irrespective of its value. Would you be able to do another pass and let me know if you don't agree with me?

Reply:
Ok I see now. I'd missed the redirectURL building directly below this. Makes sense.